src/RL/utils/geometricflux/heterogeneousgraphconv.jl

```diff
@@ -68,4 +54,50 @@ function (g::HeterogeneousGraphConv)(fg::HeterogeneousFeaturedGraph, original_fg
         g.σ.(g.weightsval * vcat(X3, H3, H1 * vartovalN) .+ g.biasval),
         fg.gf
     )
+end
+
+function (g::HeterogeneousGraphConv{<:AbstractMatrix,<:Any,meanPooling})(fgs::BatchedHeterogeneousFeaturedGraph{Float32}, original_fgs::BatchedHeterogeneousFeaturedGraph{Float32})
+    contovar, valtovar = fgs.contovar, fgs.valtovar
+    vartocon, vartoval = permutedims(contovar, [2, 1, 3]), permutedims(valtovar, [2, 1, 3])
+    H1, H2, H3 = fgs.varnf, fgs.connf, fgs.valnf
+    X1, X2, X3 = original_fgs.varnf, original_fgs.connf, original_fgs.valnf
+
+    Zygote.ignore() do
+        contovarN = contovar ./ reshape(mapslices(x -> sum(eachrow(x)), contovar, dims=[1, 2]), 1, :, size(contovar, 3))
+        valtovarN = valtovar ./ reshape(mapslices(x -> sum(eachrow(x)), valtovar, dims=[1, 2]), 1, :, size(valtovar, 3))
+        vartoconN = vartocon ./ reshape(mapslices(x -> sum(eachrow(x)), vartocon, dims=[1, 2]), 1, :, size(vartocon, 3))
+        vartovalN = vartoval ./ reshape(mapslices(x -> sum(eachrow(x)), vartoval, dims=[1, 2]), 1, :, size(vartoval, 3))
+
+        return BatchedHeterogeneousFeaturedGraph{Float32}(
+            contovar,
+            valtovar,
+            g.σ.(g.weightsvar ⊠ vcat(X1, H1, H2 ⊠ contovarN, H3 ⊠ valtovarN) .+ g.biasvar),
+            g.σ.(g.weightscon ⊠ vcat(X2, H2, H1 ⊠ vartoconN) .+ g.biascon),
+            g.σ.(g.weightsval ⊠ vcat(X3, H3, H1 ⊠ vartovalN) .+ g.biasval),
+            fgs.gf
+        )
+    end
```

Comment:
We should not forget to correct this before merging with master. As the function is not recursive, it is ok to return under `Zygote.ignore()` statement but the operations that are computed inside the return here should be copied outside thr `Zygote.ignore()` domain. We have solved the same issue on `heterogeneousgraphtransformer.jl`.

Reply:
What causes the Zygote errors is the construction of the `BatchedHeterogeneousFeaturedGraph`. It would even be safer to do separate construction of the object under Zygote.ignore() and return outside it.